token/core/fabtoken/actions.go

```diff
@@ -76,35 +95,50 @@ func (i *IssueAction) GetOutputs() []driver.Output {
 	return res
 }
 
+// IsAnonymous returns false, indicating that the identity of issuers in fabtoken
+// is revealed during issue
 func (i *IssueAction) IsAnonymous() bool {
 	return false
 }
 
+// GetIssuer returns the issuer encoded in IssueAction
 func (i *IssueAction) GetIssuer() []byte {
 	return i.Issuer
 }
 
+// GetMetadata returns nil, indicating that IssueAction in fabtoken carries no metadata
 func (i *IssueAction) GetMetadata() []byte {
 	return nil
 }
 
+// TransferAction encodes a fabtoken transfer
 type TransferAction struct {
-	Sender  view.Identity
-	Inputs  []string
+	// sender's ecdsa public key
+	// a transfer action has a single sender (//todo check is this needs to be changed)
+	Sender view.Identity
+	// identifier of token to be transferred
+	Inputs []string
+	// outputs to be created as a result of the transfer
 	Outputs []*TransferOutput
 }
 
+// Serialize marshals TransferAction
 func (t *TransferAction) Serialize() ([]byte, error) {
 	return json.Marshal(t)
 }
 
+// NumOutputs returns the number of outputs in an TransferAction
 func (t *TransferAction) NumOutputs() int {
 	return len(t.Outputs)
 }
 
+// GetSerializedOutputs returns the serialization of the outputs in a TransferAction
 func (t *TransferAction) GetSerializedOutputs() ([][]byte, error) {
 	var res [][]byte
-	for _, output := range t.Outputs {
+	for k, output := range t.Outputs {
+		if output == nil {
+			return nil, errors.Errorf("cannot serialize transfer action outputs: nil output at index [%d]", k)
+		}
 		ser, err := output.Serialize()
 		if err != nil {
 			return nil, err
```

Comment:
maybe add details to the error here as well

Reply:
done